### PR TITLE
Add slicing args defined in @listSize directive

### PR DESCRIPTION
**Pull request by ErikWittern**
Addresses https://github.com/IBM/graphql-query-generator/issues/15

Does not yet document the option, as that relies on exposing the "specification" defining the exact structure and semantics of the @listSize directive.

### CODE REVIEW

File: src/generate-query.ts [hunk 1]
```diff
@@ -447,10 +448,64 @@ function getNextNodefactor(variableValues: { [name: string]: any }): number {
   return 1
 }
 
+/**
+ * Returns the first slicing argument defined in the field's @listSize diretive,
+ * if:
+ *  - The @listSize diretive is indeed present, and defines slicing arguments
```
Comment (Alan-Cha) [inline src/generate-query.ts:454]: `diretive` -> `directive` typo

File: src/generate-query.ts [hunk 2]
```diff
@@ -447,10 +448,64 @@ function getNextNodefactor(variableValues: { [name: string]: any }): number {
...
+ *  - The requiredArguements do not already include any of the defined slicing
```
Comment (Alan-Cha) [inline src/generate-query.ts:455]: `requiredArguements` -> `requiredArguments` typo

File: src/generate-query.ts [hunk 3]
```diff
@@ -447,10 +448,64 @@ function getNextNodefactor(variableValues: { [name: string]: any }): number {
...
+ *    arguments
+ *  - The @listSize diretive doesn't also set requireOneSlicingArgument to false
```
Comment (Alan-Cha) [inline src/generate-query.ts:457]: `diretive` -> `directive` typo

File: src/generate-query.ts [hunk 4]
```diff
@@ -447,10 +448,64 @@ function getNextNodefactor(variableValues: { [name: string]: any }): number {
   return 1
 }
 
+/**
+ * Returns the first slicing argument defined in the field's @listSize diretive,
```
Comment (Alan-Cha) [inline src/generate-query.ts:452]: I think we should point to the documentation of the`@listSize` directive

File: src/generate-query.ts [hunk 5]
```diff
@@ -447,10 +448,64 @@ function getNextNodefactor(variableValues: { [name: string]: any }): number {
...
+  )
+  if (
+    typeof requireOneSlicingArg !== 'undefined' &&
+    requireOneSlicingArg.value.kind === 'BooleanValue' &&
+    requireOneSlicingArg.value.value === false
+  )
+    return null
+
+  // Return null if slicing arguments are not a list:
+  if (slicingArgumentsArg.value.kind !== 'ListValue') return null
```
Comment (Alan-Cha) [inline src/generate-query.ts:487]: Can we combine this with line 473?

File: src/generate-query.ts [hunk 6]
```diff
@@ -467,6 +524,8 @@ function getArgsAndVars(
   const requiredArguments = allArgs.filter((arg) =>
     considerArgument(arg, config)
   )
+  const missingSlicingArg = getMissingSlicingArg(requiredArguments, field)
```
Comment (Alan-Cha) [inline src/generate-query.ts:527]: Can we add a comment here that clarifies that this feature utilizes the API Connect directives?

File: test/consider-listsize-directive.test.ts [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+  const { queryDocument } = generateRandomQuery(schema, { seed: 1 })
+  const query = print(queryDocument)
+  expect(query.trim()).toEqual(
+    dedent(`query RandomQuery {
+      orders {
+        id
+        date
+      }
+    }
+    `).trim()
```
Comment (Alan-Cha) [inline test/consider-listsize-directive.test.ts:27]: Small comment but I think it is a bit odd that some of these strings start with a backtick, some of them start with a line break, some of them end with a backtick, and some of them end with a line break (and then a back tick).

**Comment by Alan-Cha** [comment]
We need to describe this in the README, no?

File: src/generate-query.ts (hunk 1 above)
Comment (ErikWittern) [reply src/generate-query.ts:454]: Will fix.

File: src/generate-query.ts (hunk 2 above)
Comment (ErikWittern) [reply src/generate-query.ts:455]: Will fix.

File: src/generate-query.ts (hunk 3 above)
Comment (ErikWittern) [reply src/generate-query.ts:457]: Will fix.

File: src/generate-query.ts (hunk 5 above)
Comment (ErikWittern) [reply src/generate-query.ts:487]: Will do.

File: src/generate-query.ts (hunk 6 above)
Comment (ErikWittern) [reply src/generate-query.ts:527]: Will add a comment.

File: src/generate-query.ts (hunk 4 above)
Comment (ErikWittern) [reply src/generate-query.ts:452]: Will add a TODO until the specification / documentation of the `@listSize` directive to point to exists.

File: test/consider-listsize-directive.test.ts (hunk 7 above)
Comment (ErikWittern) [reply test/consider-listsize-directive.test.ts:27]: Will make them all start and end with a line break, as I think it is most readable.

**Pull request merged**